Teapot/TeapotError.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+extension Teapot {
```

Comment:
Where would this be used outside of Teapot? 🤔

Reply:
I just don't get it a struct that is part of a class... That just means when you want to initialise it you have to do it from Teapot?

Reply:
Why would you be initialising it outside of Teapot, if it's an internal teapot error? 👍

Reply:
I need it to initialise it from MockedTeapot

Reply:
Also here ou init it from `NetworkResult` and than you need: 
`self = .failure(response, Teapot.TeapotError.missingImage)`
instead of otherwise just `self = .failure(response, TeapotError.missingImage)`